code_challenges.rb

```diff
@@ -9,6 +9,26 @@
 # Expected output: [-823, 7, 23]
 
 
+# was trying to take this original piece of code and try to get it to select for odd values!! This one was difficult because i spent alot of time trying to find the method that would filter out only numbers
```

Comment:
ok cool! so let's start with sorting out numbers. everything in ruby is a class. and we can run `.class` on any item in Ruby to see what it is an instance of. The numbers will be of class `Integer`. So we may have code that looks like this to get rid of non-numbers:
```
def is_odd array
  just_odds = array.select.with_index do |value, index|
    value.class == Integer
  end
end
```

Now, we want to make sure they're odd. Ruby is so awesome with all the shortcut methods! If we want to add that, our code will evolve to look like this:
```
def is_odd array
  just_odds = array.select.with_index do |value, index|
    value.class == Integer && value.odd?
  end
end
```
Now we need to sort it!
```
def is_odd array
  just_odds = array.select.with_index do |value, index|
    value.class == Integer && value.odd?
  end
  just_odds.sort
end
```
aaaaaand voila!